src/operator/webhooks/clientintents_webhook_v2alpha1.go

```diff
@@ -0,0 +1,448 @@
+	intentsList *otterizev2alpha1.ClientIntentsList) *field.Error {
+
+	desiredClientName := intentsObj.GetWorkloadName()
+	for _, existingIntent := range intentsList.Items {
+		// Deny admission if intents already exist for this client, and it's not the same object being updated
+		if existingIntent.GetWorkloadName() == desiredClientName && existingIntent.Name != intentsObj.Name && existingIntent.GetClientKind() == intentsObj.GetClientKind() {
+			return &field.Error{
+				Type:     field.ErrorTypeDuplicate,
+				Field:    "name",
+				BadValue: desiredClientName,
+				Detail: fmt.Sprintf(
+					"Intents for client %s already exist in resource %s", desiredClientName, existingIntent.Name),
+			}
+		}
+	}
+	return nil
+}
+
+// validate kubernetes kind
+func (v *IntentsValidatorV2alpha1) validateKubernetesKind(kind string) *field.Error {
+	if kind != "" && strings.ToUpper(string(kind[0])) != string(kind[0]) {
+		return &field.Error{
+			Type:   field.ErrorTypeInvalid,
+			Field:  "kind",
+			Detail: "Kubernetes Kinds must start with an uppercase letter",
+		}
+	}
+	return nil
+}
+
+// validateSpec
+func (v *IntentsValidatorV2alpha1) validateSpec(intents *otterizev2alpha1.ClientIntents) *field.Error {
+	// validate that if kind is specified, it starts with an uppercase letter
+	if err := v.validateKubernetesKind(intents.Spec.Workload.Kind); err != nil {
+		return err
+	}
+	if strings.Contains(intents.Spec.Workload.Name, ".") {
+		return &field.Error{
+			Type:   field.ErrorTypeForbidden,
+			Field:  "workload.name",
+			Detail: "Workload name should not contain '.' character",
+		}
+	}
+
+	for _, target := range intents.GetCallsList() {
+		err := v.validateIntentTarget(target)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Validate intent target
+func (v *IntentsValidatorV2alpha1) validateIntentTarget(intentTarget otterizev2alpha1.Target) *field.Error {
+	err := v.validateOnlyOneTargetFieldSet(intentTarget)
+	if err != nil {
+		return err
+	}
+	if err := v.validateKubernetesTarget(intentTarget.Kubernetes); err != nil {
+		return err
+	}
+
+	if err := v.validateServiceTarget(intentTarget.Service); err != nil {
+		return err
+	}
+
+	if err := v.validateKafkaTarget(intentTarget.Kafka); err != nil {
+		return err
+	}
+	if err := v.validateSQLTarget(intentTarget.SQL); err != nil {
+		return err
+	}
+
+	if err := v.validateAWSTarget(intentTarget.AWS); err != nil {
+		return err
+	}
+
+	if err := v.validateGCPTarget(intentTarget.GCP); err != nil {
+		return err
+	}
+
+	if err := v.validateAzureTarget(intentTarget.Azure); err != nil {
+		return err
+	}
+
+	if err := v.validateInternetTarget(intentTarget.Internet); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Validate that one and only one of the target's fields is set. Wwe do it by marshal the target to json and check there's only one key
+func (v *IntentsValidatorV2alpha1) validateOnlyOneTargetFieldSet(target otterizev2alpha1.Target) *field.Error {
```

Comment:
@orishoshan - I feel like there must be a smarter way to do it, please tell me if something comes to your mind. 
(The only requirement: if a new "target type" is introduced (rabbitMQ for example), this test will function without any changes)